### PR TITLE
Implement readonly mode

### PR DESCRIPTION
**Pull request by kostyanf14**
resolve: #90

Signed-off-by: Kostiantyn Kostiuk <kostyanf14@live.com>

### CODE REVIEW

File: src/components/comment-form.js [hunk 1]
```diff
@@ -63,38 +64,47 @@ const ResultForm = ({ user }) => {
...
-            </li>)}
-          </ul>
-        </>}
+  if (readOnly) {
+    return (
+      <div className='position-relative'>
+        {value}
       </div>
+    );
+  } else {
```
Comment (just806me) [inline src/components/comment-form.js:73]: else is not required here

File: src/components/result-form.js [hunk 2]
```diff
@@ -55,29 +56,37 @@ const ResultForm = ({ user, criterion, max }) => {
...
-          onChange={onChange}
-          onFocus={onFocus}
-          onBlur={onBlur} />
+  if (readOnly) {
+    return (
+      <div className='position-relative'>
+        {value}
       </div>
+    );
+  } else {
```
Comment (just806me) [inline src/components/result-form.js:65]: same

File: src/components/comment-form.js (hunk 1 above)
Comment (kostyanf14) [reply src/components/comment-form.js]: done

File: src/components/comment-form.js [hunk 3]
```diff
@@ -63,20 +64,29 @@ const ResultForm = ({ user }) => {
...
+  }
+
   return (
     <div className='position-relative'>
       <div className='input-group' style={{ minWidth: 400 }}>
         <TextareaAutosize
           className={inputClassName}
           value={value}
-          disabled={lockedId && !lockAcquired}
+          disabled={(lockedId && !lockAcquired) || readOnly}
```
Comment (just806me) [inline src/components/comment-form.js:81]: `readOnly` will never be true

File: src/components/comment-form.js [hunk 4]
```diff
@@ -63,20 +64,29 @@ const ResultForm = ({ user }) => {
...
           onChange={onChange}
           onFocus={onFocus}
           onBlur={onBlur} />
         {!value && !dirty && <>
           <button
             className='btn btn-outline-secondary dropdown-toggle'
             type='button'
+            disabled={(lockedId && !lockAcquired) || readOnly}
```
Comment (just806me) [inline src/components/comment-form.js:89]: `readOnly` will never be true

File: src/components/result-form.js [hunk 5]
```diff
@@ -65,7 +74,7 @@ const ResultForm = ({ user, criterion, max }) => {
           min={0}
           max={max}
           value={value}
-          disabled={lockedId && !lockAcquired}
+          disabled={(lockedId && !lockAcquired) || readOnly}
```
Comment (just806me) [inline src/components/result-form.js:77]: `readOnly` will never be true

File: src/components/comment-form.js [hunk 6]
```diff
@@ -63,20 +64,29 @@ const ResultForm = ({ user }) => {
     [user, dirty, lockAcquired]
   );
 
+  if (readOnly) {
+    return (
+      <div className='position-relative'>
+        {value}
+      </div>
+    );
```
Comment (just806me) [inline src/components/comment-form.js:72]: Relative position was used for status-notice, it is not necessary here.

```suggestion
    return <div>{value}</div>;
```

File: src/components/result-form.js [hunk 7]
```diff
@@ -55,6 +56,14 @@ const ResultForm = ({ user, criterion, max }) => {
     [user, criterion, dirty, lockAcquired]
   );
 
+  if (readOnly) {
+    return (
+      <div className='position-relative'>
+        {value}
+      </div>
+    );
```
Comment (just806me) [inline src/components/result-form.js:64]: Relative position was used for status-notice, it is not necessary here.

```suggestion
    return <div>{value}</div>;
```

**Review by just806me — CHANGES_REQUESTED** [review]

File: src/components/comment-form.js (hunk 3 above)
Comment (kostyanf14) [reply src/components/comment-form.js]: removed

File: src/components/result-form.js (hunk 2 above)
Comment (kostyanf14) [reply src/components/result-form.js]: done

File: src/components/comment-form.js (hunk 4 above)
Comment (kostyanf14) [reply src/components/comment-form.js]: removed

File: src/components/result-form.js (hunk 5 above)
Comment (kostyanf14) [reply src/components/result-form.js]: removed

File: src/components/comment-form.js (hunk 6 above)
Comment (kostyanf14) [reply src/components/comment-form.js]: removed

File: src/components/result-form.js (hunk 7 above)
Comment (kostyanf14) [reply src/components/result-form.js]: removed

**Review by just806me — APPROVED** [review]

**Pull request merged**